app.rb

```diff
@@ -0,0 +1,49 @@
+require './person'
+require './teacher'
+require './student'
+require './book'
+require './rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'List of all books:'
+    @books.each { |book| puts book.title }
+  end
+
+  def list_people
+    puts 'List of all people:'
+    @people.each { |person| puts person.name }
+  end
+
+  def create_person(name, age, type, parent_permission: true, specialization: nil)
+    if type == 'student'
+      @people << Student.new(name, age, parent_permission)
+    elsif type == 'teacher'
+      @people << Teacher.new(name, age, specialization)
+    else
+      puts 'Invalid type of person!'
+    end
+  end
+
+  def create_book(title, author)
+    @books << Book.new(title, author)
+  end
```

Comment:
- [ ] Same here, There is an error raised while trying to create a new book, Please fix that.

![image](https://user-images.githubusercontent.com/68796232/227720009-e72a0d3d-7c78-4b1e-aa5b-43c96f24352e.png)

Reply:
resolved! ✔️